tests/foreman/cli/test_hostgroup.py

```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
+
+        @id: c468fcac-9e42-4ee6-a431-abe29b6848ce
+
+        @Assert: Hostgroup should be created and has lifecycle env assigned
+
+        @BZ: 1309107
+        """
+        org = make_org()
+        lc_env = make_lifecycle_environment({'organization-id': org['id']})
+        hostgroup = make_hostgroup({
+            'lifecycle-environment': lc_env['name'],
+            'organization-ids': org['id'],
+            'lifecycle-environment-organization-id': org['id'],
+        })
+        self.assertEqual(
+            lc_env['name'],
+            hostgroup['lifecycle-environment'],
+        )
+
+    @skip_if_bug_open('bugzilla', 1309107)
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_multiple_entities(self):
+        """Check if hostgroup with multiple options can be created
+
+        @id: a3ef4f0e-971d-4307-8d0a-35103dff6586
+
+        @Assert: Hostgroup should be created and has all defined entities
+        assigned
+
+        @BZ: 1309107
+        """
+        subnet = make_subnet()
+        domain = make_domain()
+        org = make_org()
+        lc_env = make_lifecycle_environment({'organization-id': org['id']})
+        loc = make_location()
+        env = make_environment()
+        os = make_os()
+        media = make_medium()
+        arch = make_architecture()
+        puppet_proxy = Proxy.list()[0]
+        ptable = 'Kickstart default'
+        hostgroup = make_hostgroup({
+            'domain-id': domain['id'],
+            'subnet-id': subnet['id'],
+            'organization-ids': org['id'],
+            'location-ids': loc['id'],
+            'environment-id': env['id'],
+            'puppet-proxy-id': puppet_proxy['id'],
+            'puppet-ca-proxy-id': puppet_proxy['id'],
+            'operatingsystem-id': os['id'],
+            'architecture-id': arch['id'],
+            'lifecycle-environment': lc_env['name'],
+            'lifecyle-environment-organization-id': org['id'],
+            'partition-table': ptable,
+            'medium-id': media['id'],
+        })
+        self.assertEqual(domain['name'], hostgroup['domain'])
+        self.assertEqual(subnet['name'], hostgroup['subnet'])
+        self.assertEqual(org['name'], hostgroup['organization'])
+        self.assertEqual(env['name'], hostgroup['environment'])
+
+    @skip_if_bug_open('bugzilla', 1354568)
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_subnet_id(self):
+        """Check if hostgroup with invalid subnet id raises proper error
+
+        @id: c352d7ea-4fc6-4b78-863d-d3ee4c0ad439
+
+        @Assert: Proper error should be raised
+
+        @BZ: 1354568
+        """
+        subnet_id = gen_string('numeric', 4)
+        with self.assertRaises(CLIReturnCodeError) as exception:
+            HostGroup.create({
+                'name': gen_string('alpha', 10),
+                'subnet-id': subnet_id
+            })
+        self.assertIs(exception.exception.stderr, 'Could not find subnet {0}'
```

Comment:
self.assertIs(
    exception.exception.stderr, 
    'Could not find subnet {0}'.format(subnet_id)
)